numpy/ma/extras.py

```diff
@@ -638,6 +638,15 @@ def median(a, axis=None, out=None, overwrite_input=False, keepdims=False):
     else:
         return r
 
+def _median1D(data):
+    counts = filled(count(data), 0)
+    (idx, rmd) = divmod(counts, 2)
+    if rmd:
+        choice = slice(idx, idx + 1)
+    else:
+        choice = slice(idx - 1, idx + 1)
+    return data[choice].mean(0)
```

Comment:
Note the function will error if `count` is zero. Not sure what to do about empty case, but might want to raise a more informative error message than an `IndexError`.